paperqa/agents/helpers.py

```diff
@@ -0,0 +1,92 @@
+from __future__ import annotations
+
+import logging
+import re
+from typing import cast
+
+from openai import AsyncOpenAI
+from rich.table import Table
+
+from .. import OpenAILLMModel
+from ..docs import Docs
+from .models import AnswerResponse
+from .tools import get_year
+
+logger = logging.getLogger(__name__)
+
+
+async def openai_get_search_query(
+    question: str,
+    count: int,
+    template: str | None = None,
+    llm: str = "gpt-4o-mini",
+    temperature: float = 1.0,
+) -> list[str]:
+    if isinstance(template, str):
+        if not (
+            "{count}" in template and "{question}" in template and "{date}" in template
+        ):
+            logger.warning(
+                "Template does not contain {count}, {question} and {date} variables. Ignoring template."
+            )
+            template = None
+
+        else:
+            # partial formatting
+            search_prompt = template.replace(r"{date}", get_year())
```

Comment:
```suggestion
            search_prompt = template.replace("{date}", get_year())
```

Do we need the `r` here? I feel like we just don't put `f` if it's a template